### PR TITLE
fix - zx_demo

### PR DESCRIPTION
**Pull request by KetpuntoG**
This PR should fix the zx demo bug:

```
/content/src/pennylane/pennylane/tape/qscript.py:186: UserWarning: The `prep` keyword argument is being removed from `QuantumScript`, and `StatePrepBase` operations should be placed at the beginning of the `ops` list instead
```

### CODE REVIEW

File: pennylane/transforms/zx/converter.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ def processing_fn(res):
             else:
                 expanded_operations.append(op)
 
-        expanded_tape = QuantumScript(expanded_operations, mapped_tape.measurements, [])
+        expanded_tape = QuantumScript(expanded_operations, mapped_tape.measurements)
```
Comment (trbromley) [inline pennylane/transforms/zx/converter.py:339]: We probably need a test for this.

Comment (albi3ro) [reply pennylane/transforms/zx/converter.py:339]: We can edit `pytest.ini` to turn the deprecation warning into an error

Comment (albi3ro) [reply pennylane/transforms/zx/converter.py:339]: Yep. Let's place:
```
    error:The `prep` keyword argument:UserWarning:pennylane.tape.qscript
```
at the end of `tests/pytest.ini`.

Comment (KetpuntoG) [reply pennylane/transforms/zx/converter.py:339]: I have done what you say here. `pytest.ini` is a document I don't know, can you confirm that it is what you were asking for?

File: tests/pytest.ini [hunk 2]
```diff
@@ -19,3 +19,5 @@ filterwarnings =
     ignore:Casting complex values to real discards the imaginary part:UserWarning:torch.autograd
     ignore:Call to deprecated create function:DeprecationWarning
     ignore:the imp module is deprecated:DeprecationWarning
+
+error:The `prep` keyword argument:UserWarning:pennylane.tape.qscript
```
Comment (albi3ro) [inline tests/pytest.ini:23]:
```suggestion
    error:The `prep` keyword argument:UserWarning:pennylane.tape.qscript
```

**Review by albi3ro — APPROVED** [review]
👍

**Pull request closed without merging**